src/components/ConnectedFilter/utils.js

```diff
@@ -117,3 +117,24 @@ export const getFilterSections = (filters, tabsOptions, initialTabsOptions) => {
   // console.log('getFilterSections: ', sections);
   return sections;
 };
+
+export const excludeSelfFilterFromAggsData = (receivedAggsData, filterResults) => {
+  console.log(receivedAggsData);
+  if (!filterResults) return receivedAggsData;
+  const resultAggsData = {};
+  Object.keys(receivedAggsData).forEach((field) => {
+    const { histogram } = receivedAggsData[field];
+    if (!histogram) return;
+    if (field in filterResults) {
+      let resultHistogram = [];
+      if (typeof filterResults[field].selectedValues !== 'undefined') {
+        const { selectedValues } = filterResults[field];
+        resultHistogram = histogram.filter(bucket => selectedValues.includes(bucket.key));
+      }
+      resultAggsData[field] = { histogram: resultHistogram };
+    } else {
+      resultAggsData[field] = receivedAggsData[field];
+    }
+  });
+  return resultAggsData;
+};
```

Comment:
Can someone please explain why this was introduced? Wouldn't `receivedAggsData` already be based on `filterResults`, thereby making the work done in lines 129-134 effectively a no-op?

Reply:
I think it was because by default this filter generated agg queries will have `filterSelf: false` https://github.com/uc-cdis/guppy/blob/7d5a077221ad7c77360ff13837689be1d0c3d8a1/src/components/Utils/queries.js#L31, which means it will return all fields back. See also https://github.com/uc-cdis/guppy/blob/master/doc/queries.md#filterself

Reply:
@mfshao Thanks for your answer. In other words, `excludeSelfFilterFromAggsData` is intended to undo the effect of `filterSelf: false` and strictly return the filtered result.